### PR TITLE
*: update to go 1.21

### LINKED ISSUE #6923: Update the go version to 1.21
## Enhancement Task

See https://go.dev/blog/go1.21 for more info.

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: Close #6923.

### What is changed and how does it work?

```commit-message
Update to go 1.21.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
Update the go version to 1.21.
```

### CODE REVIEW

**Comment by JmPotato** [comment]
/hold

Wait for the GitHub Actions and CI updates.

**Comment by purelind** [comment]
> /hold
> 
> Wait for the GitHub Actions and CI updates.

Hi, the go version in build CI has been updated to go1.21 now.

**Review by lhy1024 — APPROVED** [review]

File: .github/workflows/tso-consistency-test.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     steps:
       - uses: actions/setup-go@v3
         with:
-          go-version: "1.20.1"
+          go-version: '1.21'
```
Comment (bufferflies) [inline .github/workflows/tso-consistency-test.yaml:13]:
```suggestion
          go-version: '1.21.0'
```

Comment (JmPotato) [reply .github/workflows/tso-consistency-test.yaml:13]: I intend to use 1.21, which is the equivalent of 1.21.*, to ensure that we can benefit from the minor version.

**Review by rleungx — APPROVED** [review]

**Pull request merged**